scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala

```diff
@@ -298,7 +303,78 @@ object node {
   // shape of this data should probably change, currently we leave it for backward compatibility
   object FragmentInputDefinition {
 
-    @JsonCodec case class FragmentParameter(name: String, typ: FragmentClazzRef)
+    @JsonCodec case class FixedExpressionValue(expression: String, label: String)
+
+    object FragmentParameterInputMode {
+
+      @JsonCodec
+      sealed trait FragmentParameterInputMode
+
+      case object InputModeAny extends FragmentParameterInputMode
+
+      @JsonCodec
+      sealed trait InputModeWithFixedList extends FragmentParameterInputMode
+
+      case object InputModeAnyWithSuggestions extends InputModeWithFixedList
+
+      case object InputModeFixedList extends InputModeWithFixedList
```

Comment:
`InputModeFixedList` and `InputModeWithFixedList` are almost identical to me. Is this "middle" trait really needed? Looks like the domain model is more complex to save us one `case` in `match`?